Model/Sync/Orders/Processor.php

```diff
@@ -483,6 +483,10 @@ public function syncOrder($order, $isYotpoSyncedOrder, $yotpoSyncedOrders)
                 ['external_ids' => $incrementId, 'entityLog' => 'orders']
             );
         } elseif ($this->isImmediateRetry($response, $this->entity, $orderId, $order->getStoreId())) {
+            $missingProducts = $this->getMissingProductIds($response);
+            if ($missingProducts) {
+                $this->catalogProcessor->removeProductFromSyncTable($missingProducts, [$storeId]);
+            }
```

Comment:
This block should be executed only if the response code that was returned is 404 (not found).

Reply:
isImmediateRetry($response, $this->entity, $orderId, $order->getStoreId())) will satisfy this (401 conditions we are already resolving in other cases, as we discussed in the call we did last week).

Also it will return from the below condition if it is not the case of 404

public function getMissingProductIdsFromNotFoundResponse($response)
    {

$messagePrefix = "products not found:";

        if (strpos($message, $messagePrefix) === false) {
            return [];
        } 
}